.github/workflows/main_workflow.yml

```diff
@@ -0,0 +1,40 @@
+# Management of the action do made during a pull request
+
+name: 'Main_workflow'
+on:
+  pull_request:
+    branches: main
+  push:
+    branches: main
+
+# Cancel existing runs 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  linter:
+    name: Linter
+    uses: ./.github/workflows/call_black.yml
+  
+  tests:
+    name: tests
+    needs: [linter]
+    uses: ./.github/workflows/call_test_package.yml
+
+  make_documentation:
+    name: Make_documentation
+    needs: [linter, tests]
+    uses: ./.github/workflows/call_build_docs.yml
```

Comment:
really nice that github allows to compose workflow like this: is this a new thing? or have I been living under a rock?

Reply:
It's a feature which was introduced in 2020.
For me, it seems that the feature is used generally by people but I like it because I feel that this gives a better way to manage the workflow.